warehouse/redshift/redshift.go

```diff
@@ -376,12 +381,33 @@ func (rs *HandleT) loadTable(tableName string, tableSchemaInUpload, tableSchemaA
 		partitionKey = column
 	}
 
+	var skipEntireTableClause string
+	if dedupWindow {
+		if _, ok := tableSchemaAfterUpload["received_at"]; ok {
+			skipEntireTableClause = fmt.Sprintf(`AND %[1]s.%[2]s.received_at > GETDATE() - INTERVAL '%d DAY'`, rs.Namespace, tableName, dedupWindowInDays)
+		}
+	}
+
 	var additionalJoinClause string
 	if tableName == warehouseutils.DiscardsTable {
 		additionalJoinClause = fmt.Sprintf(`AND _source.%[3]s = %[1]s.%[2]s.%[3]s AND _source.%[4]s = %[1]s.%[2]s.%[4]s`, rs.Namespace, tableName, "table_name", "column_name")
 	}
 
-	sqlStatement = fmt.Sprintf(`DELETE FROM %[1]s."%[2]s" using %[1]s."%[3]s" _source where (_source.%[4]s = %[1]s.%[2]s.%[4]s %[5]s)`, rs.Namespace, tableName, stagingTableName, primaryKey, additionalJoinClause)
+	sqlStatement = fmt.Sprintf(`
+		DELETE FROM 
+		  %[1]s."%[2]s" using %[1]s."%[3]s" _source 
+		where 
+		  (
+			_source.%[4]s = %[1]s.%[2]s.%[4]s %[5]s %[6]s
+		  )
+`,
```

Comment:
Can we use prepared statements for this ?

Reply:
Sure.

Reply:
Currently, it is getting hard to add since there are conditionals. Although we have a task for this in this Quarter in order to have Augmentation for Query Builder.
https://www.notion.so/rudderstacks/Use-augmentation-for-Building-up-the-Query-Builder-366c0a2f9e7348d4970b8cbee03e6958

Reply:
Also, the parameters are constants only.

Reply:
Makes sense. We have done this in jobsdb https://github.com/rudderlabs/rudder-server/blob/master/jobsdb/jobsdb.go#L2527, but I understand the readability issue. 

I would defer dealing with this as part of the task you mention